src/org/sosy_lab/java_smt/api/Model.java

```diff
@@ -49,13 +49,21 @@ public interface Model extends Evaluator, Iterable<ValueAssignment>, AutoCloseab
    *       within a quantified context, some value assignments can be missing in the iteration.
    *       Please use a direct evaluation query to get the evaluation in such a case.
    * </ul>
+   *
+   * <p>Warning: this might throw an unchecked {@link SolverException} as an extension of {@link
+   * Throwable}.
```

Comment:
This message is confusing. First, what is "an extension" of something in the context of Java? I don't remember ever hearing this term, and I would not know what you mean with that. Second, `SolverException` is a checked exception as defined by JLS 11.1.1. So "unchecked `SolverException`" is an inherent contradiction.

Reply:
I would still say it is not correct. We do not throw (as) `RuntimeException`, which I would interpret as `throw new RuntimeException(e);`.

Suggestion:
Warning: This method may throw the checked exceptions `SolverException` (in case of solver failures) and `InterruptedException` (in case of shutdown requests) although these exceptions are not declared with `throws`.

Reply:
Looks good! Thank you for the suggestion! I added it as proposed.